newIDE/app/src/MainFrame/Toolbar/PreviewAndPublishButtons.js

```diff
@@ -0,0 +1,192 @@
+  isPreviewEnabled: boolean,
+  hasPreviewsRunning: boolean,
+  previewState: PreviewState,
+  exportProject: () => void,
+  hasProject: boolean,
+|};
+
+export default function PreviewAndPublishButtons({
+  onPreviewWithoutHotReload,
+  onNetworkPreview,
+  onOpenDebugger,
+  onHotReloadPreview,
+  canDoNetworkPreview,
+  isPreviewEnabled,
+  hasPreviewsRunning,
+  previewState,
+  setPreviewOverride,
+  exportProject,
+  hasProject,
+}: PreviewAndPublishButtonsProps) {
+  const previewTooltip = hasPreviewsRunning
+    ? 'Apply changes to the running preview, right click for more'
+    : previewState.isPreviewOverriden
+    ? 'Preview is overridden, right click for more'
+    : previewState.previewExternalLayoutName
+    ? 'Launch a preview of the external layout inside the scene, right click for more'
+    : 'Launch a preview of the scene, right click for more';
+  return (
+    <ThemeConsumer>
+      {muiTheme => (
+        <React.Fragment>
+          <ElementWithMenu
+            element={
+              <ToolbarIcon
+                disabled={!isPreviewEnabled}
+                src="res/ribbon_default/bug32.png"
+                tooltip={t`Advanced preview options (debugger, network preview...)`}
+              />
+            }
+            buildMenuTemplate={(i18n: I18nType) => [
+              {
+                label: i18n._(t`Start Network Preview (Preview over WiFi/LAN)`),
+                click: onNetworkPreview,
+                enabled: canDoNetworkPreview,
+              },
+              { type: 'separator' },
+              {
+                label: i18n._(
+                  t`Start Preview with Debugger and Performance Profiler`
+                ),
+                click: onOpenDebugger,
+              },
+            ]}
+          />
+          <ElementWithMenu
+            element={
+              <FlatButton
+                onClick={onHotReloadPreview}
+                disabled={!isPreviewEnabled}
+                icon={
+                  <img
+                    alt={t`Preview`}
+                    src={
+                      hasPreviewsRunning
+                        ? 'res/ribbon_default/hotReload64.png'
+                        : previewState.isPreviewOverriden
+                        ? 'res/ribbon_default/previewOverride32.png'
+                        : 'res/ribbon_default/preview64.png'
+                    }
+                    width={32}
+                    height={32}
+                    style={{
+                      filter: !isPreviewEnabled
+                        ? 'grayscale(100%)'
+                        : muiTheme.gdevelopIconsCSSFilter,
+                    }}
+                  />
+                }
+                label={
+                  hasPreviewsRunning ? (
+                    <Trans>Update</Trans>
+                  ) : (
+                    <Trans>Preview</Trans>
+                  )
+                }
+                exceptionalTooltipForToolbar={<Trans>{previewTooltip}</Trans>}
```

Comment:
Don't use `Trans` like this. Ask @AlexandreSi if he remembers why ;) 
By doing this, you generate a translation which is `<0/>`, i.e: a unique variable to translate. Compare this with: `Hello {name}` => this will generate a translation in French of `Bonjour {name}` (and maybe a different word order in another language). But how should `{name}` be translated in French? It should not! That's the same here => you probably want to either pass an object resulting from a `t` macro (i.e: a MessageDescriptor) and you use i18n inside the button OR (better I think here), you pass something that is already considered translated (i.e: a translated string with `i18n._` or a `Trans`, though I'm not sure tooltips work with components, just with strings)